### DISCUSSION TITLE
Is it possible to deploy a Node.js server on GitHub pages?

### DISCUSSION THREAD

**Discussion by The-Best-Codes** [community/community#78878] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello, everyone!

Today I really wanted to deploy an API with Vercel, but I can't because I don't give out my phone number, even for verification purposes. I looked all over the place, but couldn't really tell if it is possible to deploy a functional Node.js API with GitHub. I have a GitHub site set up at:
[https://the-best-codes.github.io/](https://the-best-codes.github.io/?ref=github_community&site=GitHub&reason=post_comm&click=https://github.com/orgs/community/discussions/78878)
But is there a way I can make free APIs for it (preferably on GitHub or somewhere free)? 

Thanks in advance!

**Comment by Kiran1689** [comment, marked as answer]
Hi @The-Best-Codes 
If your API is serves static content (HTML, CSS, JavaScript), you can deploy it using gh pages. Because GitHub Pages is supports only static sites, and it will not support server-side code.

The best options are Vercel or Heroku..

**Reply by The-Best-Codes** [reply]
Thanks for your answer! I have tried Vercel and Heroku, but both require a phone or payment options. I'm looking for something free and less personal (but hopefully still unlimited?). Do you have any recommendations? Thanks again!

**Reply by Kiran1689** [reply]
Yeah, Heroku needs payment option.
Try vercel only it's free and use sign-up using GitHub instead email if you don't want to provide your number..
Once check it ..

**Reply by The-Best-Codes** [reply]
I did sign up with GitHub and it still wanted my phone number. 😕 

![image](https://github.com/community/community/assets/106822363/841b6654-aa30-41bb-b5f4-cc1830ed1608)

Are there any other free alternatives? Thanks for your help!

**Reply by baumamit** [reply]
https://www.netlify.com/ works fine for me

**Reply by The-Best-Codes** [reply]
Thank you for your support, everyone. I went ahead and gave Vercel my phone number. Vercel has been working awesome for me since then!

**Reply by LeushinSG1972** [reply]
https://dashboard.render.com/web/new

**Reply by The-Best-Codes** [reply]
Yup, I have also tried Render, but I mostly use serverless now. I got hooked on Vercel :melting_face:
